src/vsr/replica_test.zig

```diff
@@ -1496,45 +1496,6 @@ test "Cluster: scrub: background scrubber, fully corrupt grid" {
     try TestReplicas.expect_equal_grid(b1, b2);
 }
 
-// Compat(v0.15.3)
-test "Cluster: client: empty command=request operation=register body" {
-    const t = try TestContext.init(.{ .replica_count = 3 });
-    defer t.deinit();
-
-    // Wait for the primary to settle, since this test doesn't implement request retries.
-    t.run();
-
-    var client_bus = try t.client_bus(0);
-    defer client_bus.deinit();
-
-    var request_header = vsr.Header.Request{
-        .cluster = t.cluster.options.cluster_id,
-        .size = @sizeOf(vsr.Header),
-        .client = client_bus.client_id,
-        .request = 0,
-        .command = .request,
-        .operation = .register,
-        .release = releases[0].release,
-    };
-    request_header.set_checksum_body(&.{}); // Note the absence of a `vsr.RegisterRequest`.
-    request_header.set_checksum();
-
-    client_bus.request(t.replica(.A0).index(), &request_header, &.{});
-    t.run();
-
-    const Reply = extern struct {
-        header: vsr.Header.Reply,
-        body: vsr.RegisterResult,
-    };
-
-    const reply = std.mem.bytesAsValue(Reply, client_bus.reply.?.buffer[0..@sizeOf(Reply)]);
-    try expectEqual(reply.header.command, .reply);
-    try expectEqual(reply.header.operation, .register);
-    try expectEqual(reply.header.size, @sizeOf(Reply));
-    try expectEqual(reply.header.request, 0);
-    try expect(stdx.zeroed(std.mem.asBytes(&reply.body)));
-}
```

Comment:
Ah, maybe not remove this test altogether, but rather change the expectation? That is , assert that the client is evicted. 


That's my general rule of thumb for tests that stride versions: instead of testing just the old behavior, or just the new behavior, write the test as

```zig
for (.{.old, .new}) |behavior| {
    // normal test body //
}
```

and inside switch on `.behavior` to specifically underscore the _diff_ in the response!  

The spirit of this rule would've caught the "invalid before eviction" effect without us having to discover it by hand.